### ISSUE TITLE
inline annotations for the interfaces - what is the correct way?

### ISSUE THREAD

**Issue by vitek-rostislav** [shopsys/shopsys#1369]
### What is happening
In tests, we often access services using DI container directly - e.g.
```
/** @var \Shopsys\FrameworkBundle\Component\Domain\Domain $domain */
$domain = $this->container->get(Domain::class);
```

However, when grabbing an interface this way, we are not consistent with the annotations, you can see that often there is no @var annotation at all, or we annotate it with a particular framework implementation, like that:
```
/** @var \Shopsys\FrameworkBundle\Model\Product\ProductDataFactory $productDataFactory */
$productDataFactory = $this->getContainer()->get(ProductDataFactoryInterface::class);
```

Is there any reason why not to annotate it with
```
\Shopsys\FrameworkBundle\Model\Product\ProductDataFactoryInterface
```
directly?

The inconsistency might cause problems to annotations fixer implemented in #1344 as it has no information that it should replace 
```
\Shopsys\FrameworkBundle\Model\Product\ProductDataFactory
``` 
in the annotation with 
```
\Shopsys\ShopBundle\Model\Product\ProductDataFactory
```
Actually, the fixer looks for 
```
\Shopsys\FrameworkBundle\Model\Product\ProductDataFactoryInterface
``` 
as this is the "parent" for project class that is registered in DIC.

### Expected result
We know how to use inline annotations for interfaces and it is implemented or documented.

**Comment by jDolba** [comment]
what about not using this annotations at all?

most of the examples in `tests` (with direct access to container) might/should be covered by `.phpstorm.meta.php` file
example:
https://gist.github.com/bcremer/be2a558a452d4f45dc04994ffe7b8292

general definition in `phpstorm.meta.php` file might be 
`override(\Psr\Container\ContainerInterface::get(0), type(0));`

```
$domain = $this->container->get(\Domain::class);
// PHP STORM IDE should know now the $domain is "\Domain"
```

for static analysis (phpstan) there are already extensions (https://github.com/phpstan/phpstan-symfony/blob/master/README.md) so php-stan should be able to analyze without php-docs


so basically the questions is:
are the annotations intended for IDE? if so: is phpstorm recommended IDE? if so -> use phpstorm.meta file
